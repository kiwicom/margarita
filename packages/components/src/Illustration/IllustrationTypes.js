// @flow

import { type StylePropType } from '@kiwicom/universal-components';

type NameType =
  | 'Accommodation'
  | 'AirHelp'
  | 'AirportTransport'
  | 'AirportTransportTaxi'
  | 'AirportShuttle'
  | 'BaggageDrop'
  | 'Boarding'
  | 'BoardingPass'
  | 'BusinessTravel'
  | 'CabinBaggage'
  | 'DesktopSearch'
  | 'EnjoyApp'
  | 'Error'
  | 'FastTrack'
  | 'Feedback'
  | 'Help'
  | 'Improve'
  | 'Insurance'
  | 'InviteAFriend'
  | 'Login'
  | 'Lounge'
  | 'Mailbox'
  | 'Meal'
  | 'MobileApp'
  | 'Money'
  | 'MusicalInstruments'
  | 'NetVerify'
  | 'NoBookings'
  | 'NoFavoriteFlights'
  | 'Nomad'
  | 'NoNotification'
  | 'NoResults'
  | 'Offline'
  | 'OnlineCheckIn'
  | 'OpenSearch'
  | 'Parking'
  | 'Pets'
  | 'PlaceholderAirport'
  | 'PlaceholderDestination'
  | 'PlaceholderHotel'
  | 'PlaceholderTours'
  | 'PlaneAndMoney'
  | 'PriorityBoarding'
  | 'Rating'
  | 'ReferAFriend'
  | 'RentalCar'
  | 'Seating'
  | 'SpecialAssistance'
  | 'SportsEquipment'
  | 'Success'
  | 'Time'
  | 'TimelineBoarding'
  | 'TimelineDropBaggage'
  | 'TimelineLeave'
  | 'TimelinePick'
  | 'Tours'
  | 'Train'
  | 'TransportBus'
  | 'TransportTaxi';

export type IllustrationProps = {|
  name: NameType,
  style?: StylePropType,
|};
