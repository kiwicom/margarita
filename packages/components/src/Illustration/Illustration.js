// @flow

import React from 'react';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import { StyleSheet } from '@kiwicom/universal-components';

import { type IllustrationProps } from './IllustrationTypes';

const baseImagePath = (name, resolution: number) =>
  `//images.kiwi.com/illustrations/0x${parseInt(
    defaultTokens.heightIllustrationMedium,
    10,
  ) * resolution}/${name}.png`;

const RESOLUTIONS = [1, 2, 3];

export default function Illustration({ name, style = {} }: IllustrationProps) {
  return (
    <img
      src={baseImagePath(name, 1)}
      srcSet={RESOLUTIONS.map(
        resolution => `${baseImagePath(name, resolution)} ${resolution}x`,
      ).join(',')}
      style={{
        alignSelf: 'center',
        ...StyleSheet.flatten(style),
      }}
    />
  );
}
