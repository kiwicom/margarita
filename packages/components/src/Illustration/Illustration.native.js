// @flow

import React from 'react';
import { Image } from 'react-native';
import { StyleSheet } from '@kiwicom/universal-components';

import { type IllustrationProps } from './IllustrationTypes';
// $FlowFixMe - Cannot resolve module RelativeImageStub ( possible solution https://bit.ly/2BYfHRr )
import Boarding from './assets/mobile/Boarding.png';
// $FlowFixMe
import Error from './assets/mobile/Error.png';
// $FlowFixMe
import NoResults from './assets/mobile/NoResults.png';

const getIllustration = name => {
  switch (name) {
    case 'Boarding':
      return Boarding;
    case 'Error':
      return Error;
    case 'NoResults':
      return NoResults;
    default:
      return null;
  }
};

export default function Illustration({ name, style }: IllustrationProps) {
  const illustration = getIllustration(name);
  return (
    illustration && (
      <Image style={[styles.illustration, style]} source={illustration} />
    )
  );
}

const styles = StyleSheet.create({
  illustration: {
    alignSelf: 'center',
    resizeMode: 'contain',
  },
});
