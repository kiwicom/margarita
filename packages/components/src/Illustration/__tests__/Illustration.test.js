// @flow

import * as React from 'react';
import { shallow } from 'react-native-testing-library';

import Illustration from '../Illustration';

it('renders', () => {
  expect(shallow(<Illustration name="Boarding" />)).toMatchSnapshot();
});
