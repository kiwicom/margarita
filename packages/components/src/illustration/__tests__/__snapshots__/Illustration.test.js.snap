// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Object {
  "output": <Component
    source={1}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "resizeMode": "contain",
        },
        undefined,
      ]
    }
  />,
}
`;
